core/src/test/java/org/elasticsearch/ingest/core/IngestDocumentTests.java

```diff
@@ -969,31 +970,24 @@ public void testIngestMetadataTimestamp() throws Exception {
     public void testCopyConstructor() {
         IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random());
         IngestDocument copy = new IngestDocument(ingestDocument);
-        recursiveEqualsButNotSameCheck(ingestDocument.getSourceAndMetadata(), copy.getSourceAndMetadata());
```

Comment:
Only this test was failing, right?
Why not just change the `recursiveEqualsButNotSameCheck(...)` method to do `arrayEquals(...)` in case of arrays?